schemas/iso19139/src/main/plugin/iso19139/formatter/xsl-view/view.xsl

```diff
@@ -596,14 +596,17 @@
   <xsl:template mode="render-field"
                 match="gmd:fileIdentifier"
                 priority="100">
+    <xsl:variable name="fileIdentifier">
+      <xsl:apply-templates mode="render-value" select="*"/>
+    </xsl:variable>
     <dl>
       <dt>
         <xsl:value-of select="tr:nodeLabel(tr:create($schema), name(), null)"/>
       </dt>
       <dd>
         <xsl:apply-templates mode="render-value" select="*"/>
         <xsl:apply-templates mode="render-value" select="@*"/>
-        <a class="btn btn-default" href="{$nodeUrl}api/records/{$metadataId}/formatters/xml">
+        <a class="btn btn-default" href="{$nodeUrl}api/records/{$fileIdentifier}/formatters/xml">
```

Comment:
You can probably simply do $metadataUuid see https://github.com/geonetwork/core-geonetwork/blob/master/web/src/main/webapp/WEB-INF/data/data/formatter/xslt/render-variables.xsl#L61
and remove the variable.

Reply:
Going to change, didn't see the variable was already there.